### PR TITLE
Challenge landing page backend

### PR DESCRIPTION
**Pull request by ahmed-ben10**
- I moved all the challenge related code to the /challenge folder
- added backend route where we fetch all the challenges for the authenticated user
-  added a resource to transform the challenge attributes
- Made frontend work

### CODE REVIEW

File: packages/server/src/app/challenge/routers/challenge.router.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import express from 'express';
+import { jwtAuthHeaderValidator } from '../../auth/index.js';
+import { userChallengeController } from '../controllers/challenge.controller.js';
+import { challengeSuggestionRouter } from './challenge_suggestion.router.js';
+
+export const challengeRouter = express.Router();
+
+challengeRouter.get('/', jwtAuthHeaderValidator, async (request, response) => {
```
Comment (NoNamer777) [inline packages/server/src/app/challenge/routers/challenge.router.js:8]: I may be misremembering, but I thought we had spoken about that GET `/api/challenge/` would return all challenges (more an API end point for managing the challenges), while GET `/api/user/:userId/challenge` would return all challenges from 1 particular user?

File: packages/client/src/app/challenge/models/user_challenge.model.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  completed;
+
+  /** @type {number} */
+  user_id;
+
+  /** @type {number} */
+  challenge_id;
+
+  /** @type {number} */
+  badge_id;
```
Comment (majsa1) [inline packages/client/src/app/challenge/models/user_challenge.model.js:15]: Badge Id has been removed from user_challenge

File: packages/client/src/app/challenge/services/challenge.service.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+  /** @type {ChallengeService} */
+  static #instance;
+
+  /**
+   * @param userId {number}
+   * @return {Promise<Array>}
+   */
+  async getChallenges(userId) {
+    return await HttpRequestService.instance().getRequest(`/api/user/${userId}/challenges`);
```
Comment (majsa1) [inline packages/client/src/app/challenge/services/challenge.service.js:20]: We have user in singular and challenges in plural, let's decide on which one to use.

File: packages/server/src/app/challenge/validators/challenge.validator.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    notEmpty: { bail: true, errorMessage: 'Een startdatum is verplicht.' },
+    isDate: { bail: true, errorMessage: 'Een startdatum moet een geldige datum zijn' },
+    escape: true
+  },
+  end_date: {
+    notEmpty: { bail: true, errorMessage: 'Een einddatum is verplicht.' },
+    isDate: { bail: true, errorMessage: 'Een einddatum moet een geldige datum zijn' },
+    escape: true
+  },
+  groups: {
```
Comment (majsa1) [inline packages/server/src/app/challenge/validators/challenge.validator.js:20]: A group should be nullable?

**Review by majsa1 — CHANGES_REQUESTED** [review]
Looks nice, left some comments :-)
Can you solve the conflicts and I'll see how it looks in browser too?

File: packages/server/src/app/challenge/validators/challenge.validator.js (hunk 4 above)
Comment (ahmed-ben10) [reply packages/server/src/app/challenge/validators/challenge.validator.js:20]: Yes, because you can also have a challenge with yourself

**Review by majsa1 — APPROVED** [review]
Nice!

**Pull request merged**